File: Python_benchmark/bubble_sort_benchmark.py
import time
import random

random.seed(42)

def bubble_sort(arr):
    N = len(arr)
    for i in range(N - 1):
        swapped = False
        for j in range(N - 1 - i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:
            break

def main():
    N = int(input("Enter array size: "))
    arr = [random.randint(0, 10000) for _ in range(N)]
    
    start_time = time.perf_counter()
    bubble_sort(arr)
    end_time = time.perf_counter()
    
    print(f"Bubble Sort | Array Size: {N} | Time: {(end_time - start_time) * 1000:.2f} ms")

if __name__ == "__main__":
    main()

File: CPP_benchmark/bubble_sort_benchmark.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <chrono>
#include <random>

using namespace std;
using namespace chrono;

void bubble_sort(vector<int>& arr) {
    int N = arr.size();
    for (int i = 0; i < N - 1; i++) {
        bool swapped = false;
        for (int j = 0; j < N - 1 - i; j++) {
            if (arr[j] > arr[j + 1]) {
                swap(arr[j], arr[j + 1]);
                swapped = true;
            }
        }
        if (!swapped) break;
    }
}

int main(int argc, char* argv[]) {
    std::mt19937 rng(42);  // Fixed seed for reproducibility
    std::uniform_int_distribution<int> dist(0, 10000);
    if (argc != 2) {
        cerr << "Usage: " << argv[0] << " <array_size>" << endl;
        return 1;
    }

    int N = stoi(argv[1]);
    vector<int> arr(N);
    for (int& num : arr) num = rand();

    auto start = high_resolution_clock::now();
    bubble_sort(arr);
    auto end = high_resolution_clock::now();

    cout << "Bubble Sort | Array Size: " << N 
         << " | Time: " << duration<double, milli>(end - start).count() << " ms" << endl;
    return 0;
}
